keps/sig-node/2400-node-swap/README.md

```diff
@@ -731,11 +849,11 @@ Here are specific improvements to be made:
 
 #### GA
```

Comment:
Let's try to re-confirm beta criteria are satisfied:

Immidiately:

1. https://kubernetes.io/docs/concepts/architecture/nodes/#swap-memory doesn't have any notes about the `encrypted swap` and no examples on how to implement it.
2. Eviction: the issue https://github.com/kubernetes/kubernetes/issues/120800 is still open

Maybe we can checkbox other criterias.

Reply:
For the eviction, it will be great to have an e2e test

Reply:
>1. [kubernetes.io/docs/concepts/architecture/nodes#swap-memory](https://kubernetes.io/docs/concepts/architecture/nodes/#swap-memory) doesn't have any notes about the encrypted swap and no examples on how to implement it.

You're correct. I do mention encrypted swap in the blog-post I wrote about beta for the 1.28 release [1] though. I'll make sure to make a PR to update https://kubernetes.io/docs/concepts/architecture/nodes/#swap-memory as well.

[1] https://kubernetes.io/blog/2023/08/24/swap-linux-beta/#security-risk

> Eviction: the issue https://github.com/kubernetes/kubernetes/issues/120800 is still open

This issue is important, but pretty outdated (e.g. triaged as `triage/needs-information`) and can be closed. As said in my [comment below](https://github.com/kubernetes/enhancements/pull/4701#issuecomment-2360345618), extending evictions is planned to be dealt with in the near future with a dedicated follow-up KEP.

Saying that, it's important to note that swap works well today without these planned extensions. There is a lot of positive feedback from users that are using swap in production regarding this, and for good reasons (although I don't want to dive into too technical details here).

> For the eviction, it will be great to have an e2e test

The only problem with such a test is that it's pretty hard to do. Swap e2e tests are pretty disruptive as is, and adding evictions to that is even more difficult and flake-prone.

Reply:
Can the example of copnfiguring encrypted swap be added to the documentation?

Eviction - replied in the other comment.

Were all other graduation criteria satisfied?

Reply:
> Can the example of copnfiguring encrypted swap be added to the documentation?

Absolutely. I'll start working on it right away.
I've also added it as a GA criteria, alongside documenting the eviction limitations.

> Were all other graduation criteria satisfied?

This is the current status:
- [X] Test a wide variety of scenarios that may be affected by swap support, including tests with aggressive memory stress.
- [X] Address memory-backed backed volumes which should not have access to swap.  
- [ ] - Remove feature gate.
- [X] Exclude high-priority, static and mirrored pods from gaining access to swap.
- [ ] Add documentation regarding encrypted swap.
- [ ] Add documentation regarding limitations around evictions.

Reply:
@SergeyKanzhelev FYI that I have this ongoing PR for a blog post about swap graduating to GA on 1.32: https://github.com/kubernetes/website/pull/48099.

I will shortly add a similar PR to update the main docs.